configuration/uart.md

```diff
@@ -1,51 +1,111 @@
+The Raspberry Pi Zero, 1, 2, and 3 each contain two UARTs as follows:
 
-The default value of the `enable_uart` flag depends on the actual roles of the UARTs, so that if ttyAMA0 is assigned to the Bluetooth module, `enable_uart` defaults to 0. If the mini UART is assigned to the Bluetooth module, then `enable_uart` defaults to 1. Note that if the UARTs are reassigned using a Device Tree Overlay (see below), `enable_uart` defaults will still obey this rule.
+| Name | Type |
+|------|------|
+|UART0 |PL011 |
+|UART1 |mini UART |
 
-## Disabling Linux's use of console UART
+## Pi 4 - six UARTS
 
-In a default install of Raspbian, the primary UART (serial0) is assigned to the Linux console. Using the serial port for other purposes requires this default behaviour to be changed. On startup, `systemd` checks the Linux kernel command line for any console entries, and will use the console defined therein. To stop this behaviour, the serial console setting needs to be removed from command line.
+The Raspberry Pi 4 has four additional PL011s, which are disabled by default. The full list of UARTs on the Pi 4 is:
 
-This can be done by using the [raspi-config](raspi-config.md) utility, or manually.
-```
-sudo raspi-config
-```
-Select option 5, **Interfacing options**, then option P6, **Serial**, and select **No**. Exit raspi-config.
+| Name | Type |
+|------|------|
+|UART0 |PL011 |
+|UART1 |mini UART |
+|UART2 |PL011 |
+|UART3 |PL011 |
+|UART4 |PL011 |
+|UART5 |PL011 |
+
+## Primary UART
+
+On the Raspberry Pi, one UART is selected to be present on GPIO 14 (transmit) and 15 (receive) - this is the primary UART. By default, this will also be the UART on which a Linux console may be present. Note that GPIO 14 is pin 8 on the GPIO header, while GPIO 15 is pin 10.
+
+## Secondary UART
 
-To manually change the settings, edit the kernel command line with `sudo nano /boot/cmdline.txt`. Find the console entry that refers to the serial0 device, and remove it, including the baud rate setting. It will look something like `console=serial0,115200`. Make sure the rest of the line remains the same, as errors in this configuration can stop the Raspberry Pi from booting.
+The secondary UART is not normally present on the GPIO connector. By default, the secondary UART is connected to the Bluetooth side of the combined wireless LAN/Bluetooth controller, on models which contain this controller.
 
-Reboot the Raspberry Pi for the change to take effect.
+## Configuration
 
-## UART output on GPIO pins
+The following table summarises the default configuration of the first two UARTs:
 
-By default, the UART transmit and receive pins are on GPIO 14 and GPIO 15 respectively, which are pins 8 and 10 on the GPIO header.
+| Model | mini UART | first PL011 (UART0)|
```

Comment:
@andrum99 Does this table work if the header-row is switched to "Model, Primary UART, Secondary UART" (with the table-cells then showing either mini-UART or PL011-UART) ? Would that be less confusing or more confusing? :wink:

Reply:
I think the way it is is probably OK, although I have swapped the columns as it makes more sense to list the UARTs in order (mini UART is UART 1).